feature-libs/product-configurator/common/components/configurator-cart-entry-info/configurator-cart-entry-info.component.ts

```diff
@@ -31,9 +31,11 @@ export class ConfiguratorCartEntryInfoComponent {
    * @returns {boolean} - whether the status of configuration infos entry has status
    */
   hasStatus(item: OrderEntry): boolean {
-    return (
-      item?.configurationInfos?.length > 0 &&
-      item?.configurationInfos[0]?.status !== 'NONE'
-    );
+    const configurationInfos = item?.configurationInfos;
+
+    return configurationInfos
+      ? configurationInfos.length > 0 &&
+          configurationInfos[0]?.status !== 'NONE'
+      : false;
```

Comment:
The two added tests are already green with old version of the code, so this isn't fixing anything?
Or is this more about code-style, as the new version is a bit more explicit and maybe easier to understand.

Reply:
This fixes the strict mode compliance, but your are right, the old version also lets the tests pass (because item?.configurationInfos?.length can be compared to 0 even if configurationInfos is undefined. But besides strict mode compliance that is hard to understand, and one even doesn't notice immediately why the code lets the test pass)